src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloudRestartPreemptedIT.java

```diff
@@ -0,0 +1,139 @@
+import hudson.tasks.Builder;
+import hudson.tasks.Shell;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.java.Log;
+import org.awaitility.Awaitility;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.jvnet.hudson.test.JenkinsRule;
+
+/**
+ * Integration test suite for {@link ComputeEngineCloud}. Verifies that instances with preempted
+ * flag will be restarted when preempted.
+ */
+@Log
+public class ComputeEngineCloudRestartPreemptedIT {
+  @ClassRule
+  public static Timeout timeout = new Timeout(15 * TEST_TIMEOUT_MULTIPLIER, TimeUnit.MINUTES);
+
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+  private static ComputeClient client;
+  private static Map<String, String> label = getLabel(ComputeEngineCloudRestartPreemptedIT.class);
+  private static ComputeEngineCloud cloud;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    log.info("init");
+    initCredentials(jenkinsRule);
+    cloud = initCloud(jenkinsRule);
+    client = initClient(jenkinsRule, label, log);
+
+    InstanceConfiguration configuration =
+        instanceConfigurationBuilder()
+            .startupScript(DEB_JAVA_STARTUP_SCRIPT)
+            .numExecutorsStr(NUM_EXECUTORS)
+            .labels(LABEL)
+            .template(NULL_TEMPLATE)
+            .preemptible(true)
+            .googleLabels(label)
+            .build();
+
+    cloud.setConfigurations(Lists.newArrayList(configuration));
+  }
+
+  @AfterClass
+  public static void teardown() throws IOException {
+    teardownResources(client, label, log);
+  }
+
+  @Test
+  public void testIfNodeWasPreempted() throws Exception {
+    Collection<PlannedNode> planned = cloud.provision(new LabelAtom(LABEL), 1);
+    Iterator<PlannedNode> iterator = planned.iterator();
+    PlannedNode plannedNode = iterator.next();
+    String name = plannedNode.displayName;
+    plannedNode.future.get();
+    Node node = jenkinsRule.jenkins.getNode(name);
+
+    ComputeEngineComputer computer = (ComputeEngineComputer) node.toComputer();
+    assertTrue("Configuration was set as preemptible but saw as not", computer.getPreemptible());
+
+    FreeStyleProject project = jenkinsRule.createFreeStyleProject();
+    Builder step = new Shell("sleep 20");
+    project.getBuildersList().add(step);
+    project.setAssignedLabel(new LabelAtom(LABEL));
+    QueueTaskFuture<FreeStyleBuild> taskFuture = project.scheduleBuild2(0);
+
+    Awaitility.await()
+        .timeout(5, TimeUnit.MINUTES)
+        .until(() -> computer.getLog().contains("listening to metadata for preemption event"));
+
+    client.simulateMaintenanceEvent(PROJECT_ID, ZONE, name);
+    Awaitility.await().timeout(5, TimeUnit.MINUTES).until(computer::getPreempted);
```

Comment:
This 5 minute timeout is causing the IT to fail.

Reply:
It seems that this is system dependent, as it doesn't fail here when I run the tests.